File: samples/scala/app/controllers/HomeController.scala
package controllers

import javax.inject._

import play.api._
import play.api.mvc._
import play.api.i18n._

@Singleton
class HomeController @Inject()(val controllerComponents: ControllerComponents) extends BaseController with I18nSupport {

  def index = Action { implicit request: RequestHeader =>
    implicit val lang = messagesApi.preferred(request.acceptLanguages).lang
    Ok(views.html.index(messagesApi("index.welcome", "marcos")))
  }

}

File: samples/java/app/controllers/HomeController.java
package controllers;

import javax.inject.Inject;

import play.mvc.*;
import play.i18n.*;

import views.html.*;

public class HomeController extends Controller {

    private MessagesApi messagesApi;

    @Inject
    public HomeController(MessagesApi messagesApi) {
        this.messagesApi = messagesApi;
    }

    public Result index() {
        Lang lang = Http.Context.current().lang();
        return ok(index.render(messagesApi.get(lang, "index.welcome")));
    }

}
